### PR TITLE
[FEATURE] Add $__range(_ms/_s) builtin variables + add support for curly bracket variables + fix prefixed variables by other variable name

### LINKED ISSUE #1345: Change variable syntax from $myVar to ${myVar}
To better know the end of a variable in a text.

Ex: up{node=~"${prefix}something.*"}

### PR DESCRIPTION
**Pull request by Gladorme**
# Description
- Add builtin variables:
  - `$__range`: The range for the current dashboard in "human" readable format (1y 1w 1d 1h 1m 1s)
  - `$__range_s`: The range for the current dashboard in second
  - `$__range_ms`: The range for the current dashboard in millisecond
 - Add support for curly bracket variables: `$varName` can also be written `${varName}`
 - Fix prefixed variables by other variable name, before `$__range_ms` was parsed `3600_ms` instead of `3600000`. Because `$__range` was applied before `$__range_ms`. Now variables with longer names are applied first.

# Checklist

- [x] Pull request has a descriptive title and context useful to a reviewer.
- [x] Pull request title follows the `[<catalog_entry>] <commit message>` naming convention using one of the following `catalog_entry` values: `FEATURE`, `ENHANCEMENT`, `BUGFIX`, `BREAKINGCHANGE`, `IGNORE`.
- [x] All commits have [DCO signoffs](https://github.com/probot/dco#how-it-works).

Closes #1345

### CODE REVIEW

File: ui/core/src/model/time.ts [hunk 1]
```diff
@@ -102,6 +102,29 @@ export function isDurationString(maybeDuration: string): maybeDuration is Durati
   return DURATION_REGEX.test(maybeDuration);
 }
 
+export function formatDuration(duration: Duration): string {
```
Comment (Gladorme) [inline ui/core/src/model/time.ts:105]: Need to add test and remove months

File: ui/plugin-system/src/utils/variables.ts [hunk 2]
```diff
@@ -37,12 +44,14 @@ export function replaceTemplateVariable(text: string, varName: string, templateV
...
-  return text.replaceAll(variableTemplate, replaceString);
+  text = text.replaceAll(variableTemplate, replaceString);
+  return text.replaceAll(bracketsVariableTemplate, replaceString);
 }
 
 // This regular expression is designed to identify variable references in a template string.
 // It supports two formats for referencing variables:
 // 1. $variableName - This is a simpler format, and the regular expression captures the variable name (\w+ matches one or more word characters).
+// 2. ${variableName} - Useful for suffixed variables (.i.e: __range_ms is).
 // 2. [COMING SOON] ${variableName:value} - This is a more complex format that allows specifying a format function as well.
```
Comment (AntoineThebaud) [inline ui/plugin-system/src/utils/variables.ts:55]: 3.

File: ui/plugin-system/src/utils/variables.ts [hunk 3]
```diff
@@ -37,12 +44,14 @@ export function replaceTemplateVariable(text: string, varName: string, templateV
     replaceString = templateVariableValue;
   }
 
-  return text.replaceAll(variableTemplate, replaceString);
+  text = text.replaceAll(variableTemplate, replaceString);
+  return text.replaceAll(bracketsVariableTemplate, replaceString);
 }
 
 // This regular expression is designed to identify variable references in a template string.
```
Comment (AntoineThebaud) [inline ui/plugin-system/src/utils/variables.ts:51]: comment to update mentioning the 3 possibilities, or maybe just amend the second one to say that the second part `:value` is optional maybe.

File: ui/plugin-system/src/utils/variables.ts [hunk 4]
```diff
@@ -57,8 +66,15 @@ export const parseTemplateVariables = (text: string) => {
...
 
   while ((match = regex.exec(text)) !== null) {
-    if (match && match.length > 1 && match[1]) {
-      matches.add(match[1]);
+    if (match) {
+      if (match[1]) {
+        // \$(\w+)\
+        matches.add(match[1]);
+      }
+      if (match[2]) {
```
Comment (AntoineThebaud) [inline ui/plugin-system/src/utils/variables.ts:74]: I'd be in favor of adding an `else` here, to be more explicit about the fact that matching both is not possible

**Review by AntoineThebaud — CHANGES_REQUESTED** [review]

File: ui/plugin-system/src/utils/variables.ts (hunk 3 above)
Comment (Gladorme) [reply ui/plugin-system/src/utils/variables.ts:51]: Updated, wdyt?

**Review by AntoineThebaud — APPROVED** [review]

**Pull request merged**